rmw_fastrtps_cpp/include/rmw_fastrtps_cpp/TypeSupport_impl.h

```diff
@@ -528,6 +534,22 @@ void deserialize_array<std::string>(
     (void)call_new;
     if (member->array_size_ && !member->is_upper_bound_) {
         deser.deserializeArray(((rosidl_generator_c__String*)field)->data, member->array_size_);
+    } else {
+        std::vector<std::string> cpp_string_vector;
+        deser >> cpp_string_vector;
+
+        auto & string_array_field = *reinterpret_cast<rosidl_generator_c__String__Array *>(field);
+        if(!rosidl_generator_c__String__Array__init(&string_array_field, cpp_string_vector.size())) {
+            printf("unable to initialize rosidl_generator_c__String array\n");
```

Comment:
All error messages should go to `stderr`. But since the exception is thrown anyway I would say the error message is redundant and can be removed.